### ISSUE TITLE
Generic simplifyGOFromMultipleLists for gene sets databases other than GO?

### ISSUE THREAD

**Issue by krassowski** [jokergoo/simplifyEnrichment#49]
Thank you for this great package, I love the word cloud idea, and love the utilities to compare clustering results!

I wonder if it would make sense to create a generic version of `simplifyGOFromMultipleLists()`, possibly named `simplifyEnrichmentFromMultipleLists()` to match how `simplifyEnrichment()` generalises `simplifyGO()`.

Implementation-wise it could be as simple as moving the current function to a non-exported, shared function and allowing to pass a custom callback to it; assuming it would be named `simplify_function`, it would be:
- `simplify_function=simplifyGO` for `simplifyGOFromMultipleLists`, and
-  `simplify_function=simplifyEnrichment` for `simplifyEnrichmentFromMultipleLists`

I already solved it for myself by using `Heatmap()` directly with `anno_word_cloud()` as an annotation, but this could benefit others.

**Comment by krassowski** [comment]
Skimming the code, it would also help if the the general expression used to match `go_id_column` would get exposed in the shared private function so it would be `"^GO:\\d+$"` for `simplifyGO` but possibly a user-provided pattern for `simplifyEnrichmentFromMultipleLists`. By default it could much some common identifiers like KEGG, Reactome and possibly others from MSigDB (which already get special treatment in this package).

If this sounds like a good idea I am happy to create a PR at some time.

**Comment by jokergoo** [comment]
Yes, `simplifyEnrichmentFromMultipleLists()` must be implemented.

To set the generic function, yes, `simplify*()` function should be implemented as object-oriented since they do the same analysis but on different similarity matrices. However, the thing is the inputs for all these functions are all in "matrix" class, so I cannot really use the S3 or S4 OO to implement it. Also I have a plan to specifically treat pathways, thus we will have another `simplifyPathway()`. But indeed, all these functions have quite big overlap and they must be simplified.
